src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -28,8 +29,7 @@ protected AbstractPackage()
         protected override void Initialize()
         {
             base.Initialize();
-
-            var defaultForegroundThreadData = ForegroundThreadData.CreateDefault();
+            var defaultForegroundThreadData = ForegroundThreadData.CreateDefault(defaultKind: ForcedByPackageInitialize);
```

Comment:
Wouldn't mind comments on this, and the case below.  Something about us assumgin that when our package is initialized, it must be on the Foreground thread.